vortex-array/src/array/bool/compute/min_max.rs

```diff
@@ -0,0 +1,58 @@
+use std::ops::BitAnd;
+
+use vortex_dtype::DType;
+use vortex_dtype::Nullability::NonNullable;
+use vortex_error::VortexResult;
+use vortex_mask::Mask;
+use vortex_scalar::Scalar;
+
+use crate::array::{BoolArray, BoolEncoding};
+use crate::compute::{MinMaxFn, MinMaxResult};
+use crate::stats::{Precision, Stat};
+
+impl MinMaxFn<BoolArray> for BoolEncoding {
+    fn min_max(&self, array: &BoolArray) -> VortexResult<Option<MinMaxResult>> {
+        let min = array
```

Comment:
I think delegating to stats like this should live in the entry point function? So:
* Call compute::min_max
* Compute fn calls statistics::get to check for pre-existing stats.
* Compute fn calls min/max compute to populate missing value(s).
* Compute fn updates min/max statistic